server/server.go

```diff
@@ -151,15 +182,23 @@ func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
 	}
 
 	var err error
-	if cfg.Socket != "" {
-		if s.listener, err = net.Listen("unix", cfg.Socket); err == nil {
-			log.Infof("Server is running MySQL Protocol through Socket [%s]", cfg.Socket)
-		}
-	} else {
+
+	if s.cfg.Host != "" && s.cfg.Port != 0 {
```

Comment:
An empty Host can be valid for the Listen API.
A zero port is also valid to the Listen API. Its practical use seems limited, but we should note all changes in behavior.

Reply:
The configuration system defaults a Host and Port.  There needs to be a use case to disable tcp/ip, so in practical terms it is `./tidb-server -host '' -socket /tmp/tidb.sock`   It can be documented..